coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -725,40 +725,23 @@ private static Schema setupNestedNamespace(@Nonnull Schema schema, @Nonnull Stri
       case MAP:
         Schema valueSchema = schema.getValueType();
         Schema valueSchemaWithNestedNamespace = setupNestedNamespace(valueSchema, namespace);
-        Schema mapSchema = Schema.createMap(valueSchemaWithNestedNamespace);
-
-        return mapSchema;
+        return Schema.createMap(valueSchemaWithNestedNamespace);
       case ARRAY:
         Schema elementSchema = schema.getElementType();
         Schema elementSchemaWithNestedNamespace = setupNestedNamespace(elementSchema, namespace);
-        Schema arraySchema = Schema.createArray(elementSchemaWithNestedNamespace);
-
-        return arraySchema;
+        return Schema.createArray(elementSchemaWithNestedNamespace);
       case ENUM:
-        Schema enumSchemaWithNestedNamespace =
-            Schema.createEnum(schema.getName(), schema.getDoc(), namespace, schema.getEnumSymbols());
-
-        return enumSchemaWithNestedNamespace;
+        return Schema.createEnum(schema.getName(), schema.getDoc(), namespace, schema.getEnumSymbols());
       case RECORD:
-        Schema recordSchema = setupNestedNamespaceForRecord(schema, namespace);
-        return recordSchema;
+        return setupNestedNamespaceForRecord(schema, namespace);
       case UNION:
         List<Schema> types = new ArrayList<>();
-        if (isNullableType(schema)) {
-          Schema otherType = getOtherTypeFromNullableType(schema);
-          Schema otherTypeWithNestedNamespace = setupNestedNamespace(otherType, namespace);
-          Schema nullSchema = Schema.create(Schema.Type.NULL);
-          types.add(nullSchema);
-          types.add(otherTypeWithNestedNamespace);
-        } else {
-          for (Schema type : schema.getTypes()) {
-            Schema typeWithNestNamespace = setupNestedNamespace(type, namespace);
-            types.add(typeWithNestNamespace);
-          }
-        }
-        Schema unionSchema = Schema.createUnion(types);
 
-        return unionSchema;
+        for (Schema type : schema.getTypes()) {
+          Schema typeWithNestNamespace = setupNestedNamespace(type, namespace);
+          types.add(typeWithNestNamespace);
+        }
```

Comment:
Nice simplification! @funcheetah do you remember any reason why we did not go it this way in the first place?

Reply:
I think it is to be consistent with recommended order of optional field in Avro
`(Note that when a [default value](https://avro.apache.org/docs/current/spec.html#schema_record) is specified for a record field whose type is a union, the type of the default value must match the first element of the union. Thus, for unions containing "null", the "null" is usually listed first, since the default value of such unions is typically null.)` as specified in Avro spec: https://avro.apache.org/docs/current/spec.html#Unions

Reply:
The way I am reading this is that the whole paragraph is qualified by having a default value. If not, I am not sure if they recommendation applies to begin with.

Reply:
Yeah, I agree. However, consider this case, underlying file schema has [a, b], table schema evolves to [a, b, c]. c has to be `[type, null] default: value of type` or `[null, type]` to work. By putting `null` first in optional schema makes it more robust.